CMakeLists.txt

```diff
@@ -57,11 +57,27 @@ else()
   set(MENDER_USE_NLOHMANN_JSON 0)
 endif()
 
-add_library(mender_compiler_flags INTERFACE)
-target_compile_features(mender_compiler_flags INTERFACE cxx_std_11)
+#---------------------------------------------------------------------------------------------------
+# When deciding which one(s) of these to include, the general rule is this:
+#
+# If it's a cross platform component, add this:
+#   target_link_libraries(<TARGET> PUBLIC crossplatform_compiler_flags)
+#
+# If it's a platform specific component, with a cross platform interface, add this:
+#   target_link_libraries(<TARGET> PRIVATE platform_compiler_flags)
+#   target_link_libraries(<TARGET> INTERFACE crossplatform_compiler_flags)
+
+add_library(crossplatform_compiler_flags INTERFACE)
+target_compile_features(crossplatform_compiler_flags INTERFACE cxx_std_11)
+target_compile_options(crossplatform_compiler_flags INTERFACE -std=c++11)
```

Comment:
The important word in that sentence is "if". It doesn't add it for me because my compiler defaults to providing it. But then it also provides more. And we need to match c++11 exactly.

Reply:
So it would add it for `c++20` because it's not used by default these days? This is more confusing than beneficial, I can tell you that.

Reply:
I haven't tried, but I assume so, yes. It's just testing for minimal capability, and adding what's necessary for that.